src/plugins/analysis/ipc/docker/ipc_analyzer/ipc_analyzer.py

```diff
@@ -353,6 +390,35 @@ def get_fstring_from_call(ghidra_analysis, key_string, call_args, func, call, so
     return filter_relevant_indices(start, arg_values, indices, format_types)
 
 
+def find_version_strings(input_data, ghidra_analysis, result_path):
+    """
+    :type input_data: dict
+    :param ghidra_analysis: instance of GhidraAnalysis
+    :param result_path: the path of the output data file
+    :type result_path: str
+    :rtype: int
+    """
+    mode = input_data.get('mode')
+    if mode == 'format_string':
+        key_string_list = input_data.get('key_string_list')
+        if not key_string_list:
+            print("Error: key_string_list not found.")
+            return 1
+        result_list = resolve_version_format_string(ghidra_analysis, set(key_string_list))
+    elif mode == 'version_function':
+        # the elf file contains a special function just for returning its version
+        function_name = input_data.get('function_name')
+        result_list = find_function_ref_strings(function_name)
+        if not function_name:
```

Comment:
Why check this after the `find_function_ref_strings` call and not before?

Reply:
good question... -> changed